ui/app/src/components/ViewVersionDialog/ViewField.tsx

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (C) 2007-2024 Crafter Software Corporation. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as published by
+ * the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ContentTypeField } from '../../models';
+import Typography from '@mui/material/Typography';
+import { MonacoWrapper } from '../MonacoWrapper';
+import Box from '@mui/material/Box';
+import AsyncVideoPlayer from '../AsyncVideoPlayer';
+import { asLocalizedDateTime, convertTimeToTimezone } from '../../utils/datetime';
+import Tooltip from '@mui/material/Tooltip';
+import { FormattedMessage } from 'react-intl';
+import Button from '@mui/material/Button';
+import React, { useState } from 'react';
+import useItemsByPath from '../../hooks/useItemsByPath';
+import useLocale from '../../hooks/useLocale';
+import FieldVersionToolbar from '../CompareVersionsDialog/FieldVersionToolbar';
+import { fromString, serialize } from '../../utils/xml';
+import useMonacoOptions from '../../hooks/useMonacoOptions';
+import { nnou } from '../../utils/object';
+import StateItem from '../CompareVersionsDialog/FieldsTypesDiffViews/StateItem';
+import { ViewVersionDialogProps } from './utils';
+import ContentInstance from '../../models/ContentInstance';
+import useSelection from '../../hooks/useSelection';
+
+interface ViewFieldProps {
+  content: any;
+  field: ContentTypeField;
+  xml?: string;
+  showToolbar?: boolean;
+  contentTypeFields: ContentTypeField[];
+  onSelectField?(field: ContentTypeField): void;
+  setViewSubDialogState?(props: Partial<ViewVersionDialogProps>): void;
+}
+
+const typesRenderMap = {
+  text: MonacoWrapper,
+  textarea: MonacoWrapper,
+  html: MonacoWrapper,
+  'node-selector': DefaultView,
+  'checkbox-group': DefaultView,
+  repeat: DefaultView,
+  image: DefaultView,
+  'video-picker': DefaultView,
+  time: DefaultView,
+  'date-time': DefaultView,
+  boolean: DefaultView,
+  'numeric-input': DefaultView,
+  dropdown: DefaultView
+};
+
+interface DefaultViewProps {
+  field: ContentTypeField;
+  contentA: any;
+  xml: string;
+  setViewSubDialogState?(props: Partial<ViewVersionDialogProps>): void;
+}
+
+function DefaultView(props: DefaultViewProps) {
```

Comment:
Is the default component used for item versions view, since we get the render component from a map, and there are some field types that are simple to render, I didn't create a component for each one. (Thinking on the future override posibility from external sources).